src/oatpp-openssl/client/ConnectionProvider.cpp

```diff
@@ -125,6 +125,7 @@ provider::ResourceHandle<data::stream::IOStream> ConnectionProvider::get(){
   SSL_set_mode(ssl, SSL_MODE_ENABLE_PARTIAL_WRITE);
   SSL_set_connect_state(ssl);
   SSL_set_tlsext_host_name(ssl, host->c_str());
+  SSL_set1_host(ssl, host->c_str());
```

Comment:
Set the host name for validation.  This triggers the validation of the name against the SAN in the certificate.